clap_complete/src/dynamic/completer.rs

```diff
@@ -76,8 +76,61 @@ pub fn complete(
         } else if arg.is_escape() {
             is_escaped = true;
             state = ParseState::ValueDone;
-        } else if let Some(_long) = arg.to_long() {
-        } else if let Some(_short) = arg.to_short() {
+        } else if let Some((flag, value)) = arg.to_long() {
+            if let Ok(flag) = flag {
+                state = if let Some(opt) = current_cmd.get_arguments().find(|a| {
+                    a.get_long_and_visible_aliases()
+                        .map_or(false, |v| v.into_iter().find(|s| *s == flag).is_some())
+                }) {
+                    match opt.get_action() {
+                        clap::ArgAction::Set | clap::ArgAction::Append => {
+                            if value.is_some() {
+                                ParseState::ValueDone
+                            } else {
+                                ParseState::Opt(opt.clone())
+                            }
+                        }
+                        _ => {
+                            if value.is_some() {
+                                ParseState::ValueDone
+                            } else {
+                                ParseState::ValueDone
+                            }
+                        }
+                    }
+                } else {
+                    ParseState::ValueDone
+                };
+            }
+        } else if let Some(short) = arg.to_short() {
+            let mut short = short.clone();
+            let opt = short.next_flag();
```

Comment:
Don't we need to loop through to check the status of the last short?

Reply:
Will there be cases like `-abcd bar`? If so, we need to loop through to check the status of the last short. If not,  i think that we don't need to loop.

Reply:
Write a quick clap program and check

Reply:
```rust=
let mut cmd = Command::new("dynamic")
        .arg(
            clap::Arg::new("input")
                .long("input")
                .short('i')
                .value_hint(clap::ValueHint::FilePath),
        )
        .arg(
            clap::Arg::new("format")
                .long("format")
                .short('F')
                .value_parser(["json", "yaml", "toml"]),
        )
        .arg(
            clap::Arg::new("count")
                .long("count")
                .short('c')
                .action(clap::ArgAction::Count),
        )
        .args_conflicts_with_subcommands(true);
```

```
dynamic -ciF [TAB]
```
Should we consider `F` as the value of `-i`, or should we complete value of `-F`?

Or, when we're parsing short flags and we hit a short flag with an `ArgAction` of `Set` or `Append`, should we treat all the following characters as its value, or keep parsing them as separate short flags?

Reply:
I meant check how clap parses it and that will be the answer for `dynamic -ciF [TAB]` and `dynamic -cF [TAB]`.